### PR TITLE
[iOS] remove some legacy code

### PR DESCRIPTION
**Pull request by kambala-decapitator**
## Description
This PR removes runtime checks for class/method presence that now always evaluate to `true` and pre-iOS 6 orientation code.

## Motivation and Context
The less legacy and unused code, the better :)

## How Has This Been Tested?
Verified that iOS build succeeds and app still works.

## Types of change
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [X] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by Rechi** [comment]
`CDarwinUtils::DeviceHasRetina` function is now unused and can be removed. Then `getIosPlatform` function also will be unused and can be removed.

\
There is some iOS7 checking left which could be removed too.
https://github.com/xbmc/xbmc/blob/4398f988e1577495ab6a9ba1e43e6bd2ac65f465/xbmc/platform/darwin/ios/XBMCApplication.mm#L152

**Comment by kambala-decapitator** [comment]
@Rechi thanks for the input, will adjust.

@Memphiz is there any apparent reason that you decided to use method swizzling instead of simply subclassing `UIApplication`? https://github.com/xbmc/xbmc/commit/270077a4d237fba74ae84880c311126ded35c924

**Comment by Memphiz** [comment]
Yep hooking was needed to get the cursor key events from Bluetooth keyboards - apple changed behavior in ios7 and later that resulted in those events not being forwarded into derived classes.

File: xbmc/platform/darwin/ios/IOSEAGLView.mm [hunk 1]
```diff
@@ -88,44 +88,13 @@ - (void)layoutSubviews
...
-}
-
 - (void) setScreen:(UIScreen *)screen withFrameBufferResize:(BOOL)resize
 {
   CGFloat scaleFactor = 1.0;
   CAEAGLLayer *eaglLayer = (CAEAGLLayer *)[self layer];
 
   currentScreen = screen;
-  scaleFactor = [self getScreenScale: currentScreen];
+  scaleFactor = [currentScreen scale];
```
Comment (Memphiz) [inline xbmc/platform/darwin/ios/IOSEAGLView.mm]: You can’t do that ... the hardcoded List is needed because the scale method will give wrong results in the case that we compile with an not up to date iOS SDK and run on new devices that were released after the used SDK.
In case of the newer iPads when compiling with Xcode 9 for example - scale will return 1 instead of 2 or 3.
So we use that hardcoded type list and scales to support full retina solution on newer devices.
Alternative would be to always bump to newest Xcode and sdk whenever apple releases new devices.
I can‘t approve removing this. ;)

**Comment by Memphiz** [comment]
This touches a lot and might break one of the following:
Background control of Kodi playing music from control center
Proper retina resolution on the supported devices (mostly the newer ones)
Bluetooth Keyboard support

All this has to be tested when compiled with our used Xcode / SDK combination.

File: xbmc/platform/darwin/ios/IOSEAGLView.mm (hunk 1 above)
Comment (kambala-decapitator) [reply xbmc/platform/darwin/ios/IOSEAGLView.mm]: I see now. Thank you for the explanation! Will revert.

**Comment by kambala-decapitator** [comment]
> Background control of Kodi playing music from control center
nothing to break here

> Proper retina resolution on the supported devices (mostly the newer ones)
agree, will revert

> Bluetooth Keyboard support
you mean in case the private API `-handleKeyUIEvent:` is silently removed?

**Comment by Memphiz** [comment]
For the keyboard - in your question about deriving from uiapplication it sounded like you removed the hooking. As you didn’t remove it - you are right that nothing should break on that front.

**Comment by kambala-decapitator** [comment]
no, the `-handleKeyUIEvent:` hook is still there, I only removed the `-sendEvent:` one.

File: xbmc/platform/darwin/ios/XBMCApplication.mm [hunk 2]
```diff
@@ -197,7 +189,7 @@ static void XBMCsendEvent(id _self, SEL _cmd, UIEvent *event)
         int idx = GSEVENTKEY_KEYCODE;
```
Comment (Rechi) [inline xbmc/platform/darwin/ios/XBMCApplication.mm]: This value will never be used. I would remove `GSEVENTKEY_KEYCODE_IOS7` define, update the value of `GSEVENTKEY_KEYCODE` and remove the else below.

File: xbmc/platform/darwin/ios/IOSEAGLView.mm (hunk 1 above)
Comment (Memphiz) [reply xbmc/platform/darwin/ios/IOSEAGLView.mm]: Once that one commit is dropped i am in general fine with this.

Comment (kambala-decapitator) [reply xbmc/platform/darwin/ios/IOSEAGLView.mm]: Maybe using (nativeScale)[https://developer.apple.com/documentation/uikit/uiscreen/1617825-nativescale] would work? Can't test atm, will check later.

Comment (Memphiz) [reply xbmc/platform/darwin/ios/IOSEAGLView.mm]: You can believe me that I tested the obvious solutions back then. The behavior I described was even documented by apple. Those Methode return compatibility values when devices are to new for the used SDK.

**Comment by kambala-decapitator** [comment]
reverted `scale` change, only removed checking for the `-scale` selector.

**Comment by Memphiz** [comment]
Fine by me - thx much.

File: xbmc/platform/darwin/DarwinUtils.mm [hunk 3]
```diff
@@ -241,7 +241,7 @@ enum iosPlatform getIosPlatform()
   static enum iosPlatform platform = iDeviceUnknown;
 
 #if defined(TARGET_DARWIN_IOS)
-  if( platform == iDeviceUnknown )
+  if (platform == iDeviceUnknown)
```
Comment (Rechi) [inline xbmc/platform/darwin/DarwinUtils.mm]: unrelated change

File: xbmc/platform/darwin/DarwinUtils.mm [hunk 4]
```diff
@@ -114,24 +114,19 @@
 const char* CDarwinUtils::getIosPlatformString(void)
 {
   static std::string iOSPlatformString;
-  if (iOSPlatformString.empty())
-  {
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
```
Comment (Rechi) [inline xbmc/platform/darwin/DarwinUtils.mm]: Have you thought about using `std::call_once` ?

File: xbmc/platform/darwin/DarwinUtils.mm [hunk 5]
```diff
@@ -407,25 +398,23 @@ enum iosPlatform getIosPlatform()
...
+  });
   return rootFolder.c_str();
 }
 
 bool CDarwinUtils::IsIosSandboxed(void)
 {
-  static int ret = -1;
-  if (ret == -1)
-  {
+  static int ret = 0;
```
Comment (Rechi) [inline xbmc/platform/darwin/DarwinUtils.mm]: With the `dispatch_once` (or `std::call_once`) change this can be changed to bool.

File: xbmc/platform/darwin/ios/XBMCApplication.mm [hunk 6]
```diff
@@ -191,11 +185,11 @@ static void XBMCsendEvent(id _self, SEL _cmd, UIEvent *event)
       if (eventType == GSEVENT_TYPE_KEYUP)
       {
         // support 32 and 64bit arm here...
-        int idx = GSEVENTKEY_KEYCODE;
+        int idx;
```
Comment (Rechi) [inline xbmc/platform/darwin/ios/XBMCApplication.mm]: assign either `GSEVENTKEY_KEYCODE_32_BIT` or `GSEVENTKEY_KEYCODE_64_BIT` and remove one of the if/else assignments.

File: xbmc/platform/darwin/DarwinUtils.mm (hunk 4 above)
Comment (kambala-decapitator) [reply xbmc/platform/darwin/DarwinUtils.mm]: didn't even know about it :) Too used to objc/GCD. Will change.

File: xbmc/platform/darwin/ios/XBMCApplication.mm (hunk 6 above)
Comment (kambala-decapitator) [reply xbmc/platform/darwin/ios/XBMCApplication.mm]: I think it's even possible to use conditional compilation here with `#if TARGET_RT_64_BIT`

> TARGET_RT_64_BIT        - Generated code uses 64-bit pointers

**Comment by Rechi** [comment]
wrong revision was build
jenkins build this please

**Comment by kambala-decapitator** [comment]
anything else I should do/modify here?

**Comment by kambala-decapitator** [comment]
@Rechi @Memphiz what is blocking this to be merged?

**Comment by Memphiz** [comment]
If comments from @Rechi are resolved this can go in.

**Comment by kambala-decapitator** [comment]
yes, they are. Also marked conversations as Resolved.

**Comment by Memphiz** [comment]
Thx
